tests/test_config.py

```diff
@@ -227,3 +227,73 @@ def test_worker_config_precedence(tmp_path, test_input, expected_exclusions):
 def test_config_data_transclusion():
     config = copier.Worker("tests/demo_transclude/demo")
     assert config.all_exclusions == ("exclude1", "exclude2")
+
+
+@pytest.mark.parametrize(
+    "user_defaults, expected",
+    [
+        # Nothing overridden.
+        (
+            {},
+            dedent(
+            """\
+            A string: lorem ipsum
+            A number: 12345
+            A boolean: True
+            A list: one, two, three
+            """
+            )
+        ),
+        (
+            {
+                "a_string": "foo",
+                "a_number": 42,
+                "a_boolean": False,
+                "a_list": ["four", "five", "six"],
+            },
+            dedent(
+            """\
+            A string: foo
+            A number: 42
+            A boolean: False
+            A list: four, five, six
+            """
+            )
+        )
+    ]
+)
+def test_user_defaults(tmp_path_factory, user_defaults, expected):
```

Comment:
This test is a bit incomplete. For example, there are bugs above that are not detected. Namely: 

- The test should pass data directly, asserting that init data gets more priority than user defaults.

- Also it should  update a project and assert that the last answers are used instead of user defaults.

Reply:
* Added `data` to this test case as well as changes suggested. Interesting, it didn't seem to make a difference in whether the test passed or not so I'm still not exercising the ordering changes.
* Any prior art in the existing tests to point me towards would be appreciated here.